### PR TITLE
feat: new analytic UI style

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

1. new analytic UI style
2. fix QuickSight layout
3. add new api for analytics select project with apps

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [x] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: frontend/src/pages/analytics/analytics-utils.ts [hunk 1]
```diff
@@ -160,8 +161,8 @@ export const validRetentionAnalyticsItem = (item: IRetentionAnalyticsItem) => {
 
 export const validConditionItemType = (condition: IConditionItemType) => {
   return (
-    condition.conditionOption?.value !== null &&
-    condition.conditionOption?.value?.length !== 0
+    condition.conditionOption !== null &&
```
Comment (zxkane) [inline frontend/src/pages/analytics/analytics-utils.ts:164]: how about `condition.conditionOption?.value?.trim().length > 0`

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 2]
```diff
@@ -586,385 +471,374 @@ const AnalyticsPath: React.FC = () => {
...
+                ) : null}
+              </div>
+            </SpaceBetween>
+            <SpaceBetween direction="vertical" size="xs">
+              <Box variant="awsui-key-label">
+                {t('analytics:labels.nodeType')}{' '}
+                <Popover
+                  triggerType="custom"
+                  size="small"
+                  content="This instance contains insufficient memory. Stop the instance, choose a different instance type with more memory, and restart it."
```
Comment (zxkane) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:588]: What is this?

**Review by zxkane — CHANGES_REQUESTED** [review]
update end-to-end tests in PR description

File: frontend/src/pages/analytics/dashboard/create/CreateDashboard.tsx [hunk 3]
```diff
@@ -220,6 +226,12 @@ const CreateDashboard: React.FC<CreateDashboardProps> = (
 
             <FormField
               label={t('analytics:dashboard.createSheets')}
+              info={
+                <Popover triggerType="custom" content={t('analytics:information.dashboardSheetInfo')}>
+            <Link variant="info">Info</Link>
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/dashboard/create/CreateDashboard.tsx:231]: translate `info`

File: frontend/src/pages/analytics/event/AnalyticsEvent.tsx [hunk 4]
```diff
@@ -393,314 +298,298 @@ const AnalyticsEvent: React.FC = () => {
...
+        <Container
+          header={
+            <Header
+              variant="h2"
+              info={
+                <Popover
+                  triggerType="custom"
+                  content={t('analytics:information.eventInfo')}
+                >
+                  <Link variant="info">Info</Link>
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/event/AnalyticsEvent.tsx:312]: translate

File: frontend/src/pages/analytics/explore/AnalyticsExplore.tsx [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+            <ContentLayout
+              header={
+                <Header
+                  variant="h1"
+                  info={
+                    <Popover
+                      triggerType="custom"
+                      content={t('analytics:information.exploreInfo')}
+                    >
+                      <Link variant="info">Info</Link>
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/explore/AnalyticsExplore.tsx:266]: translate

File: frontend/src/pages/analytics/explore/AnalyticsExplore.tsx [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+                    </Popover>
+                  }
+                  description={t('analytics:explore.description')}
+                  actions={
+                    <FormField
+                      label={
+                        <>
+                          {t('analytics:explore.analyticsModel')} &nbsp;
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/explore/AnalyticsExplore.tsx:274]: remove &nbsp;

File: frontend/src/pages/analytics/explore/AnalyticsExplore.tsx [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+                          <Popover
+                            triggerType="custom"
+                            content={t(
+                              'analytics:information.analyticsModelInfo'
+                            )}
+                          >
+                            <Link variant="info">Info</Link>
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/explore/AnalyticsExplore.tsx:281]: translate

File: frontend/src/pages/analytics/metadata/table/MetadataTableHeader.tsx [hunk 8]
```diff
@@ -11,53 +11,31 @@
...
-          <Button
-            data-testid="header-btn-create"
-            variant="primary"
-            onClick={onRefreshButtonClick}
-          >
-            {refreshButtonText}
-          </Button>
-        </SpaceBetween>
+        <Popover triggerType="custom" content={infoContent}>
+          <Link variant="info">Info</Link>
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/metadata/table/MetadataTableHeader.tsx:37]: translate

File: frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributesTable.tsx [hunk 9]
```diff
@@ -245,72 +242,38 @@ const MetadataUserAttributes: React.FC = () => {
...
-      }
-      splitPanelOpen={showSplit}
-      onSplitPanelToggle={(e) => {
-        setShowSplit(e.detail.open);
+    <MetadataTable
+      resourceName="UserAttribute"
+      tableColumnDefinitions={COLUMN_DEFINITIONS}
+      tableContentDisplay={CONTENT_DISPLAY}
+      tableFilteringProperties={FILTERING_PROPERTIES}
+      tableI18nStrings={{
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributesTable.tsx:250]: I see the similar I18N String above, can we define these as a common constant?

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 10]
```diff
@@ -586,385 +471,374 @@ const AnalyticsPath: React.FC = () => {
...
+        <Container
+          header={
+            <Header
+              variant="h2"
+              info={
+                <Popover
+                  triggerType="custom"
+                  content={t('analytics:information.pathInfo')}
+                >
+                  <Link variant="info">Info</Link>
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:485]: translate

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 11]
```diff
@@ -586,385 +471,374 @@ const AnalyticsPath: React.FC = () => {
...
+          }
+        >
+          <div className="cs-analytics-config">
+            <SpaceBetween direction="vertical" size="xs">
+              <Box variant="awsui-key-label">
+                {t('analytics:labels.metrics')}{' '}
+                <Popover
+                  triggerType="custom"
+                  size="small"
+                  content="This instance contains insufficient memory. Stop the instance, choose a different instance type with more memory, and restart it."
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:520]: translate

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 12]
```diff
@@ -586,385 +471,374 @@ const AnalyticsPath: React.FC = () => {
...
+              }
+            >
+              {t('analytics:explore.pathAnalysis')}
+            </Header>
+          }
+        >
+          <div className="cs-analytics-config">
+            <SpaceBetween direction="vertical" size="xs">
+              <Box variant="awsui-key-label">
+                {t('analytics:labels.metrics')}{' '}
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:516]: remove {" "}

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 13]
```diff
@@ -586,385 +471,374 @@ const AnalyticsPath: React.FC = () => {
...
+                />
+              </div>
+            </SpaceBetween>
+            <SpaceBetween direction="vertical" size="xs">
+              <Box variant="awsui-key-label">
+                {t('analytics:labels.sessionDefinition')}{' '}
+                <Popover
+                  triggerType="custom"
+                  size="small"
+                  content="This instance contains insufficient memory. Stop the instance, choose a different instance type with more memory, and restart it."
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:541]: translate

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 14]
```diff
@@ -586,385 +471,374 @@ const AnalyticsPath: React.FC = () => {
...
+                  options={computeMethodOptions}
+                  onChange={(event) => {
+                    setSelectedMetric(event.detail.selectedOption);
+                  }}
+                />
+              </div>
+            </SpaceBetween>
+            <SpaceBetween direction="vertical" size="xs">
+              <Box variant="awsui-key-label">
+                {t('analytics:labels.sessionDefinition')}{' '}
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:537]: remove {" "}

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx [hunk 15]
```diff
@@ -586,385 +471,374 @@ const AnalyticsPath: React.FC = () => {
...
-                  iconName="search"
-                  onClick={clickPreview}
-                  loading={loadingData}
+                  </>
+                ) : null}
+              </div>
+            </SpaceBetween>
+            <SpaceBetween direction="vertical" size="xs">
+              <Box variant="awsui-key-label">
+                {t('analytics:labels.nodeType')}{' '}
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/path/AnalyticsPath.tsx:584]: remove {" "}

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx (hunk 2 above)
Comment (chenhaiyun) [reply frontend/src/pages/analytics/path/AnalyticsPath.tsx:588]: translate

File: frontend/src/pages/analytics/realtime/AnalyticsRealtime.tsx [hunk 16]
```diff
@@ -11,38 +11,215 @@
...
+                <SpaceBetween size="m">
+                  <Header
+                    variant="h1"
+                    description={t('analytics:realtime.description')}
+                    info={
+                      <Popover
+                        triggerType="custom"
+                        content={t('analytics:information.realtimeInfo')}
+                      >
+                        <Link variant="info">Info</Link>
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/realtime/AnalyticsRealtime.tsx:141]: translate

File: frontend/src/pages/analytics/retention/AnalyticsRetention.tsx [hunk 17]
```diff
@@ -369,419 +273,428 @@ const AnalyticsRetention: React.FC = () => {
...
+              {t('analytics:explore.retentionAnalysis')}
+            </Header>
+          }
+        >
+          <ColumnLayout columns={3} variant="text-grid">
+            <SpaceBetween direction="vertical" size="l">
+              <div>
                 <Box variant="awsui-key-label">
-                  {t('analytics:labels.dateRange')}
+                  {t('analytics:labels.associateParameter')}{' '}
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/retention/AnalyticsRetention.tsx:319]: remove {" "}

File: frontend/src/pages/analytics/retention/AnalyticsRetention.tsx [hunk 18]
```diff
@@ -369,419 +273,428 @@ const AnalyticsRetention: React.FC = () => {
...
+                  <Popover
+                    triggerType="custom"
+                    size="small"
+                    content="This instance contains insufficient memory. Stop the instance, choose a different instance type with more memory, and restart it."
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/retention/AnalyticsRetention.tsx:323]: translate

File: frontend/src/components/common/ExtendIcon.tsx [hunk 19]
```diff
@@ -0,0 +1,151 @@
...
+        return (
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            width="16"
+            height="16"
+            fill="currentColor"
+            className="bi bi-kanban"
+            viewBox="0 0 16 16"
+          >
+            <path d="M13.5 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1h-11a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h11zm-11-1a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h11a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2h-11z" />
```
[the remark this answers is not in the document]
Comment (tyyzqmf) [reply frontend/src/components/common/ExtendIcon.tsx:36]: https://icons.getbootstrap.com/icons MIT

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx (hunk 2 above)
Comment (tyyzqmf) [reply frontend/src/pages/analytics/path/AnalyticsPath.tsx:588]: These locations are reserved for prompt information, and @luorobin-a2z  can edit at[ i18n files](https://github.com/awslabs/clickstream-analytics-on-aws/pull/241/files#diff-e54d97e6bd16f002cbf98607fce7aa874d5d0d7c011011abce5c9a00cf4dae22R240).

File: frontend/public/locales/zh/analytics.json [hunk 20]
```diff
@@ -197,8 +226,36 @@
...
+    "realtimeStopped": "实时流已停止",
+    "realtimeStarted": "实时流已启动"
   },
   "emptyData": "无数据",
-  "emptyDataMessage": "请先配置分析指标"
+  "emptyDataMessage": "请配置要查询的指标",
+  "noDataAvailableTitle": "没有可用于分析的点击流数据",
+  "noDataAvailableMessage": "该解决方案尚未找到任何具有点击流数据的项目，请与您的系统管理员联系以开始收集数据。",
+  "information": {
+    "dashboardsInfo": "dashboardsInfo",
```
[the remark this answers is not in the document]
Comment (tyyzqmf) [reply frontend/public/locales/zh/analytics.json:241]: Robin will fill this info.

File: frontend/src/apis/analytics.ts [hunk 21]
```diff
@@ -52,11 +52,12 @@ export const deleteAnalyticsDashboard = async (
...
   projectId: string,
   appId: string,
-  dashboardId: string
+  dashboardId: string,
+  allowedDomain: string
 ) => {
   const result: any = await apiRequest(
     'get',
-    `/project/${projectId}/${appId}/dashboard/${dashboardId}`
+    `/project/${projectId}/${appId}/dashboard/${dashboardId}?allowedDomain=${allowedDomain}`
```
[the remark this answers is not in the document]
Comment (tyyzqmf) [reply frontend/src/apis/analytics.ts]: [CloudFront removes the header.](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorCustomOrigin.html)

If add this header we need use by Cloudfront function like [this blog.](https://saturncloud.io/blog/what-is-amazon-cloudfront-and-how-it-removes-referer-headers/)

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx (hunk 2 above)
Comment (zxkane) [reply frontend/src/pages/analytics/path/AnalyticsPath.tsx]: Will update them in this PR or subsequent one?

File: frontend/src/pages/analytics/dashboard/detail/AnalyticsDashboardDetail.tsx [hunk 22]
```diff
@@ -11,42 +11,32 @@
...
-          });
-        };
-        embedDashboard();
-      }
-    } catch (error) {
-      console.log(error);
-    }
-  };
+  const [dashboardEmbedUrl, setDashboardEmbedUrl] = useState('');
+  const [dashboard, setDashboard] = useState({} as IAnalyticsDashboard);
```
[the remark this answers is not in the document]
Comment (tyyzqmf) [reply frontend/src/pages/analytics/dashboard/detail/AnalyticsDashboardDetail.tsx:39]: There is no default value

File: frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributesTable.tsx (hunk 9 above)
Comment (tyyzqmf) [reply frontend/src/pages/analytics/metadata/user-attributes/MetadataUserAttributesTable.tsx:250]: The problem is that can not use i18n in constant file.

File: frontend/src/pages/analytics/path/AnalyticsPath.tsx (hunk 2 above)
Comment (tyyzqmf) [reply frontend/src/pages/analytics/path/AnalyticsPath.tsx]: Edit prompt information in other PR.

**Review by zxkane — APPROVED** [review]

**Review by zxkane — CHANGES_REQUESTED** [review]
- resolve conflicts locally

File: frontend/src/components/layouts/Navigation.tsx [hunk 23]
```diff
@@ -39,6 +39,30 @@ const Navigation: React.FC<INavigationProps> = (props: INavigationProps) => {
...
+          header={
+            t('analytics:information.introducingAnalyticsHeader') ??
+            'Introducing analytics'
+          }
+          content={
+            <div>
+              {t('analytics:information.introducingAnalyticsContent')}
+              <br />
+              <Link href="#" variant="primary">
+                Learn more.
```
Comment (chenhaiyun) [inline frontend/src/components/layouts/Navigation.tsx:57]: i18n? And the `Learn more` Link?

Comment (tyyzqmf) [reply frontend/src/components/layouts/Navigation.tsx:57]: analytics studio docs link will add later.

File: frontend/src/components/layouts/Navigation.tsx [hunk 24]
```diff
@@ -39,6 +39,30 @@ const Navigation: React.FC<INavigationProps> = (props: INavigationProps) => {
...
+            <div>
+              {t('analytics:information.introducingAnalyticsContent')}
+              <br />
+              <Link href="#" variant="primary">
+                {t('learnMore')}
+              </Link>
+            </div>
+          }
+        >
+          <Link variant="info"> New</Link>
```
Comment (zxkane) [inline frontend/src/components/layouts/Navigation.tsx:62]: create i18n text for **New**?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: frontend/src/components/layouts/Navigation.tsx (hunk 24 above)
Comment (tyyzqmf) [reply frontend/src/components/layouts/Navigation.tsx]: created.

**Review by zxkane — APPROVED** [review]

**Review by chenhaiyun — APPROVED** [review]

**Review by chenhaiyun — APPROVED** [review]

**Pull request merged**